src/test/java/org/jenkinsci/plugins/gitclient/GitAPITest.java

```diff
@@ -58,7 +58,7 @@
  */
 
 @RunWith(Parameterized.class)
-public class GitAPITest {
+public abstract class GitAPITest {
```

Comment:
I'd rather we not change this parameterized test class that checks command line git and two variants of JGit into an abstract class.  If you find that you truly need an abstract class, create a new class to receive those tests that need the abstract class.

Reply:
Thanks @MarkEWaite for the quick review, my need was to convert this class to an abstract class so that its able to host test methods that consumes abstract methods. 

> If you find that you truly need an abstract class, create a new class to receive those tests that need the abstract class.

I get now have the way to go based on the guidance.